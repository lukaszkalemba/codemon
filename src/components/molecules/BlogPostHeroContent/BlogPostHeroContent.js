import React from 'react';
import PropTypes from 'prop-types';
import Container from 'components/particles/Container';
import Heading from 'components/atoms/Heading';
import S from './BlogPostHeroContent.styles';

const BlogPostHeroContent = ({ title, description, animations }) => {
  return (
    <Container axis="both">
      <Heading>{title}</Heading>
      <S.Description
        variants={animations.descriptionVariants}
        initial="initial"
        animate="animate"
        data-testid="blog-post-hero-description"
      >
        {description}
      </S.Description>
    </Container>
  );
};

BlogPostHeroContent.defaultProps = {
  animations: {},
};

BlogPostHeroContent.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  animations: PropTypes.shape({
    descriptionVariants: PropTypes.shape({}),
  }),
};

export default BlogPostHeroContent;
