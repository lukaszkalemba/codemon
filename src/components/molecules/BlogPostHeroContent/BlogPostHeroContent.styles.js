import styled from 'styled-components';
import { motion } from 'framer-motion';

const Paragraph = styled(motion.p)`
  font-size: ${({ theme }) => theme.font.size.xxs};
  line-height: 25px;
  max-width: 450px;

  ${({ theme }) => theme.media.tablet`
    font-size: ${theme.font.size.xs};
    margin-left: 30px;
  `}

  ${({ theme }) => theme.media.laptop`
    max-width: 600px;
  `}
`;

export default { Paragraph };
