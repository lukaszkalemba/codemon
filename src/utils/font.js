export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
