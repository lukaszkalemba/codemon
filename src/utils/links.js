export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
