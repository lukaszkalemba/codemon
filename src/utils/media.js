export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
