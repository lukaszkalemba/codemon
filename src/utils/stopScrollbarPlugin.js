import { ScrollbarPlugin } from 'smooth-scrollbar';

class stopScrollbarPlugin extends ScrollbarPlugin {
  static pluginName = 'stopScrollbar';

  static defaultOptions = {
    open: false,
  };

  transformDelta(delta) {
    return this.options.open ? { x: 0, y: 0 } : delta;
  }
}

export default stopScrollbarPlugin;
