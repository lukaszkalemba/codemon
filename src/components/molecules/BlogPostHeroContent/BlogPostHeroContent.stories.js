import React from 'react';
import { text, withKnobs } from '@storybook/addon-knobs';
import BlogPostHeroContent from './BlogPostHeroContent';

const animations = {
  paragraphVariants: {
    initial: { opacity: 0, x: -30 },
    animate: {
      opacity: 0.75,
      x: 0,
      transition: { duration: 0.75, delay: 0.25 },
    },
  },
};

export const blogPostHeroContent = () => (
  <BlogPostHeroContent
    title={text('Blog Post Hero Content Title', 'Lorem ipsum.')}
    description={text(
      'Blog Post Hero Content Description',
      'Lorem ipsum dolor sit amet consectetur adipisicing elit. Aliquid laudantium qui aliquam odio quidem ipsum! Vitae quaerat praesentium cumque quo quasi magnam ex provident sunt vero, ducimus libero, dolores ipsum.'
    )}
    animations={animations}
  />
);

export default {
  component: BlogPostHeroContent,
  decorators: [withKnobs],
  title: 'Molecules|BlogPostHeroContent',
};
