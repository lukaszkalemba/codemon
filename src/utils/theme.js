import color from 'utils/color';
import font from 'utils/font';
import media from 'utils/media';
import zindex from 'utils/zindex';

export default {
  color,
  font,
  media,
  zindex,
};
