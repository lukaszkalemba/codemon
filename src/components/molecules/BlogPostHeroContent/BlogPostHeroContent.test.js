import React from 'react';
import { render } from 'utils/test-utils';
import '__mocks__/intersectionObserver';
import BlogPostHeroContent from './BlogPostHeroContent';

describe('<BlogPostHeroContent />', () => {
  it('renders correctly and matches snapshot', () => {
    const fakeTitle = 'fake blog post hero title';
    const fakeDescription =
      'Lorem ipsum dolor sit amet consectetur adipisicing elit.';

    const { container, getByRole, getByTestId } = render(
      <BlogPostHeroContent title={fakeTitle} description={fakeDescription} />
    );

    const title = getByRole('heading');
    const description = getByTestId('blog-post-hero-description');

    expect(title).toHaveTextContent(fakeTitle);
    expect(description).toHaveTextContent(fakeDescription);

    expect(container.firstChild).toMatchSnapshot();
  });
});
