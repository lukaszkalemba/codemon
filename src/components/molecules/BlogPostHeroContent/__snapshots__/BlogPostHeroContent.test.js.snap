// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPostHeroContent /> renders correctly and matches snapshot 1`] = `
<div
  class="sc-AxjAm gYmeLx"
>
  <h2
    class="sc-AxiKw kKlpRG"
    data-testid="heading"
    style="opacity: 0; transform: translateX(40px) translateZ(0);"
  >
    fake blog post hero title
  </h2>
  <p
    class="sc-AxhCb rhJVw"
    data-testid="blog-post-hero-description"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit.
  </p>
</div>
`;
